esp8266-fastled-webserver/Fields.cpp

```diff
@@ -151,6 +152,28 @@ String getCoolLikeIncandescent() {
 String getName() {
   return nameString;
 }
+String getUtcOffsetIndex() {
+  return String(utcOffsetIndex);
+}
+
+String setUtcOffsetIndex(uint8_t value)
+{
+  utcOffsetIndex = value;
+  if (utcOffsetIndex > 104) utcOffsetIndex = 104;
+  int utcOffsetInHours = map(utcOffsetIndex, 0, 104, -12, 14);
+  utcOffsetInSeconds = utcOffsetInHours * 60 * 60;
```

Comment:
`MapFloat()` appears to be a 3rd party library?  It doesn't appear to be in the base ESP8266 package.

Can we make this simpler / more explicit?

```c++
// these constants make clear the design being used
const int32_t UTC_OFFSET_MINIMUM_MINUTES = ((int32_t)-12) * 60; // corresponds to index 0
const int32_t UTC_OFFSET_INCREMENT_MINUTES = 15; // each higher index increments by this amount

// the calculations are then quite simple:
// first, figure out how many minutes above the minimum
int32_t tmp = utcOffsetIndex * UTC_OFFSET_INCREMENT_MINUTES;
// then, add that to the minimum value
tmp = MINIMUM_UTC_OFFSET_MINUTES + tmp;
// finally, convert to seconds
utcOffsetInSeconds = tmpMinutes * 60;
```

Very plain, not fancy, but easy to read and understand.
What do you think?

Reply:
mapfloat is here, in the same file: https://github.com/jasoncoon/esp8266-fastled-webserver/pull/213/files#diff-e76bdb54bdf4f249d4ac5eaa843480c5d1ca09b14f677ae5abc69bb116e825b3R33-R36

Reply:
First, thank you for helping me understand the use of the indices.  It makes sense given the fields interactions.

I will approve the PR, even if using `mapfloat()`, as it's correct code.  Because asked, I will strongly recommend to avoid floating point when integer math is available (or really, anytime it's possible to avoid(*)).  I provide the example integer math solution for your consideration.

Reply:
I like your solution better, will get it switched. Thank you for the feedback!

Reply:
Done, thanks!  Also, sorry, I should have included those explanations in the description of the PR. 😆